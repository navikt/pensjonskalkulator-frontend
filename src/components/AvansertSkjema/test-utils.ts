export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
