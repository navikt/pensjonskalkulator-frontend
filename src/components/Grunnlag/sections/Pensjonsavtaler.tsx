import React from 'react'

import { BodyLong, BodyShort, Link } from '@navikt/ds-react'
import clsx from 'clsx'

import { AccordionItem } from '@/components/components/AccordionItem'
import { formatAsDecimal } from '@/utils/currency'
import { capitalize } from '@/utils/string'

import { SectionContent } from './components/SectionContent'
import { SectionHeader } from './components/SectionHeader'
import { groupPensjonsavtalerByType } from './Pensjonsavtaler-utils'

import styles from './Pensjonsavtaler.module.scss'

interface Props {
  pensjonsavtaler: Pensjonsavtale[]
}

export function Pensjonsavtaler({ pensjonsavtaler }: Props) {
  if (pensjonsavtaler.length === 0) {
    return null
  }

  return (
    <AccordionItem name="Pensjonsavtaler" data-testid="pensjonsavtaler">
      <SectionHeader
        label="Pensjonsavtaler"
        value={`${pensjonsavtaler.length}`}
      />
      <SectionContent>
        <table className={styles.tabell}>
          <thead>
            <tr>
              <th className={styles.tabellHeader}>Pensjonsavtale</th>
              <th
                className={clsx(
                  styles.tabellHeader,
                  styles.tabellHeader__Right
                )}
              >
                Årlig utbetaling
              </th>
            </tr>
          </thead>
          <tbody>
            {Object.entries(groupPensjonsavtalerByType(pensjonsavtaler)).map(
              ([avtaleType, avtaler]) => (
                <React.Fragment key={avtaleType}>
                  <tr>
                    <td colSpan={2} className={styles.tabellCell}>
                      <BodyShort className={styles.mellomtittel}>
                        {capitalize(avtaleType)}
                      </BodyShort>
                    </td>
                  </tr>
                  {avtaler.map((avtale, i) => (
                    <tr key={i}>
                      <td className={styles.tabellCell}>
                        <BodyShort>Fra {avtale.produktbetegnelse}</BodyShort>
                        <BodyShort className={styles.utbetaling}>
                          {avtale.utbetalingsperiode.sluttAlder
                            ? `Utbetales fra ${avtale.utbetalingsperiode.startAlder} år til ${avtale.utbetalingsperiode.sluttAlder} år`
                            : `Livsvarig utbetaling fra ${avtale.utbetalingsperiode.startAlder} år`}
                        </BodyShort>
                      </td>
                      <td
                        className={clsx(
                          styles.tabellCell,
                          styles.tabellCell__Right
                        )}
                      >
                        <BodyShort>
                          {formatAsDecimal(
                            avtale.utbetalingsperiode.aarligUtbetaling
                          )}
                          kr
                        </BodyShort>
                      </td>
                    </tr>
                  ))}
                </React.Fragment>
              )
            )}
          </tbody>
        </table>
        <BodyLong className={styles.paragraph} size="small">
          Alle avtaler i privat sektor er hentet fra{' '}
          <Link href="https://norskpensjon.no/">Norsk Pensjon</Link>. Du kan ha
          andre avtaler enn det som finnes i Norsk Pensjon. Kontakt aktuell
          pensjonsordning.
        </BodyLong>
      </SectionContent>
    </AccordionItem>
  )
}
