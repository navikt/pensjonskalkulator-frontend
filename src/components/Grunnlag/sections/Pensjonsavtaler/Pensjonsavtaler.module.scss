@import '../../../../scss/variables';

.tabell {
  margin-bottom: var(--a-spacing-4);
  width: 100%;
  border-spacing: 0;
  padding: 0;

  @media (min-width: $a-breakpoint-md) {
    width: auto;
  }

  &Header {
    font-size: var(--a-font-size-heading-small);
    line-height: var(--a-font-line-height-heading-small);
    font-weight: var(--a-font-weight-bold);
    text-align: left;
    padding: var(--a-spacing-2) 0 var(--a-spacing-2) 0;

    &__Right {
      text-align: right;
      padding-left: var(--a-spacing-2);
    }
  }

  &Cell {
    padding: 0 0 var(--a-spacing-2);

    &__Right {
      text-align: right;
      vertical-align: top;
      padding-left: var(--a-spacing-2);
    }
  }
}

.mellomtittel {
  padding-top: var(--a-spacing-4);
  font-weight: var(--a-font-weight-bold);
}

.utbetaling {
  font-size: var(--a-font-size-medium);
}

.error {
  display: flex;

  &Icon {
    flex-basis: 1.5rem;
    color: var(--a-orange-600);
    vertical-align: top;
    margin-right: var(--a-spacing-2);
  }

  &Text {
    flex-basis: 100%;
  }
}

.paragraph {
  margin-bottom: var(--a-spacing-4);
}
