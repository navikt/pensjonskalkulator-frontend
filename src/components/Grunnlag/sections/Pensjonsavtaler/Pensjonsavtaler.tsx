import React from 'react'
import { useNavigate } from 'react-router-dom'

import { ExclamationmarkTriangleFillIcon } from '@navikt/aksel-icons'
import { Accordion, BodyLong, BodyShort, Link } from '@navikt/ds-react'
import clsx from 'clsx'

import { SectionContent } from '../components/SectionContent'
import { SectionHeader } from '../components/SectionHeader'
import { usePensjonsavtalerQuery } from '@/state/api/apiSlice'
import { generatePensjonsavtalerRequestBody } from '@/state/api/utils'
import { useAppSelector } from '@/state/hooks'
import { selectSamtykke } from '@/state/userInput/selectors'
import { selectCurrentSimulation } from '@/state/userInput/selectors'
import { formatAsDecimal } from '@/utils/currency'
import { capitalize } from '@/utils/string'

import {
  groupPensjonsavtalerByType,
  getPensjonsavtalerTittel,
  getMaanedString,
} from './utils'

import styles from './Pensjonsavtaler.module.scss'

export function Pensjonsavtaler() {
  const navigate = useNavigate()
  const harSamtykket = useAppSelector(selectSamtykke)
  const { startAlder, startMaaned } = useAppSelector(selectCurrentSimulation)

  const {
    data: pensjonsavtaler,
    isLoading,
    isError,
    isSuccess,
  } = usePensjonsavtalerQuery(
    generatePensjonsavtalerRequestBody({
      aar: startAlder ?? 0,
      maaned: startMaaned ?? 0,
    }),
    {
      skip: !harSamtykket || !startAlder,
    }
  )

  // PEK-97 TODO håndtere delvis reponse fra backend
  return (
    <Accordion.Item data-testid="pensjonsavtaler">
      <SectionHeader
        label="Pensjonsavtaler"
        isLoading={isLoading}
        value={getPensjonsavtalerTittel(
          !!harSamtykket,
          isError,
          `${pensjonsavtaler?.length}`
        )}
      />
      <SectionContent>
        {!harSamtykket && (
          <BodyLong>
            Du har ikke samtykket til å hente inn pensjonsavtaler om
            tjenestepensjon.{' '}
            <Link onClick={() => navigate('/start')}>
              Start en ny beregning
            </Link>{' '}
            dersom du ønsker å få dette i beregningen.
          </BodyLong>
        )}
        {isError && (
          <div className={styles.error}>
            <ExclamationmarkTriangleFillIcon
              className={styles.errorIcon}
              fontSize="1.5rem"
            />
            <BodyLong className={styles.errorText}>
              Vi klarte ikke å hente pensjonsavtalene dine fra Norsk Pensjon.
              Prøv igjen senere.
            </BodyLong>
          </div>
        )}
        {harSamtykket && isSuccess && (
          <>
            <table className={styles.tabell}>
              <thead>
                <tr>
                  <th className={styles.tabellHeader}>Pensjonsavtale</th>
                  <th
                    className={clsx(
                      styles.tabellHeader,
                      styles.tabellHeader__Right
                    )}
                  >
                    Årlig beløp
                  </th>
                </tr>
              </thead>
              <tbody>
                {Object.entries(
                  groupPensjonsavtalerByType(pensjonsavtaler)
                ).map(([avtaleType, avtaler]) => (
                  <React.Fragment key={avtaleType}>
                    <tr>
                      <td colSpan={2} className={styles.tabellCell}>
                        <BodyShort className={styles.mellomtittel}>
                          {capitalize(avtaleType)}
                        </BodyShort>
                      </td>
                    </tr>
                    {avtaler.map((avtale) => (
                      <tr key={avtale.key}>
                        <td className={styles.tabellCell}>
                          <BodyShort>Fra {avtale.produktbetegnelse}</BodyShort>

                          {avtale.utbetalingsperioder.map(
                            (utbetalingsperiode) => {
                              return (
                                <BodyShort className={styles.utbetaling}>
                                  {utbetalingsperiode.sluttAlder
                                    ? `${formatAsDecimal(
                                        utbetalingsperiode.aarligUtbetaling
                                      )} kr utbetales fra
                                  ${
                                    utbetalingsperiode.startAlder
                                  } år ${getMaanedString(
                                    utbetalingsperiode.startMaaned
                                  )} til ${
                                    utbetalingsperiode.sluttAlder
                                  } år ${getMaanedString(
                                    utbetalingsperiode.sluttMaaned
                                  )}.`
                                    : `Livsvarig utbetaling fra ${
                                        utbetalingsperiode.startAlder
                                      } år ${getMaanedString(
                                        utbetalingsperiode.startMaaned
                                      )}.`}
                                </BodyShort>
                              )
                            }
                          )}
                        </td>
                        <td
                          className={clsx(
                            styles.tabellCell,
                            styles.tabellCell__Right
                          )}
                        >
                          <BodyShort>
                            {`${formatAsDecimal(
                              avtale.utbetalingsperioder.reduce(function (
                                acc,
                                v
                              ) {
                                return acc + v.aarligUtbetaling
                              }, 0)
                            )} kr`}
                          </BodyShort>
                        </td>
                      </tr>
                    ))}
                  </React.Fragment>
                ))}
              </tbody>
            </table>
            <BodyLong className={styles.paragraph} size="small">
              Alle avtaler i privat sektor er hentet fra{' '}
              <Link href="https://norskpensjon.no/">Norsk Pensjon</Link>. Du kan
              ha andre avtaler enn det som finnes i Norsk Pensjon. Kontakt
              aktuell pensjonsordning.
            </BodyLong>
          </>
        )}
      </SectionContent>
    </Accordion.Item>
  )
}
