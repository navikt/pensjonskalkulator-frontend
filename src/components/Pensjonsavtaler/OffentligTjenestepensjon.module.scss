@import '../../scss/variables';

.info {
  display: flex;

  &Icon {
    flex-basis: 1.5rem;
    vertical-align: top;
    margin-right: var(--a-spacing-2);

    &__orange {
      color: var(--a-orange-600);
    }

    &__blue {
      color: var(--a-deepblue-600);
    }
  }

  &Text {
    flex-basis: 100%;
  }
}
