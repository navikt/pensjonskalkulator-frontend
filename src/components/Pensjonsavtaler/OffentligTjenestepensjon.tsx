import React from 'react'
import { FormattedMessage, useIntl } from 'react-intl'

import { ExclamationmarkTriangleFillIcon } from '@navikt/aksel-icons'
import { BodyLong, Heading, HeadingProps } from '@navikt/ds-react'

import { Divider } from '@/components/common/Divider'
import { Loader } from '@/components/common/Loader'
import { useGetTpoMedlemskapQuery } from '@/state/api/apiSlice'

import styles from './OffentligTjenestepensjon.module.scss'

export const OffentligTjenestepensjon = (props: {
  headingLevel: HeadingProps['level']
  showDivider?: boolean
}) => {
  const { headingLevel, showDivider } = props
  const intl = useIntl()

  const { data: tpoMedlemskap, isError, isLoading } = useGetTpoMedlemskapQuery()

  if (isLoading) {
    return (
      <Loader
        data-testid="tpo-loader"
        size="3xlarge"
        title={intl.formatMessage({
          id: 'beregning.loading',
        })}
      />
    )
  }

  if (
    !isLoading &&
    !isError &&
    tpoMedlemskap?.harTjenestepensjonsforhold === false
  ) {
    return
  }

  return (
    <>
      {showDivider && <Divider noMargin />}
      <Heading id="tpo-heading" level={headingLevel} size="small">
        {intl.formatMessage({ id: 'pensjonsavtaler.tpo.title' })}
      </Heading>
      <div className={styles.info}>
        <ExclamationmarkTriangleFillIcon
          className={`${styles.infoIcon} ${styles.infoIcon__orange}`}
          fontSize="1.5rem"
          aria-hidden
        />
        <BodyLong className={styles.infoText}>
          {isError && <FormattedMessage id="pensjonsavtaler.tpo.error" />}
          {!isError && tpoMedlemskap?.harTjenestepensjonsforhold && (
            <FormattedMessage id="pensjonsavtaler.tpo.er_medlem" />
          )}
        </BodyLong>
      </div>
    </>
  )
}
