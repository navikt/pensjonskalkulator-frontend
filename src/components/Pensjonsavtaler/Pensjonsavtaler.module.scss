@use '../../scss/variables';

.section {
  display: flex;
  flex-direction: column;
  gap: var(--a-spacing-4);

  @media (min-width: variables.$a-breakpoint-lg) {
    max-width: variables.$container-max-width;
  }

  &Header {
    font-weight: normal;
    white-space: nowrap;
    vertical-align: middle;
  }
}

.footnote {
  padding-top: var(--a-spacing-4);
  margin-top: var(--a-spacing-4);
  width: 100%;
  border-top: 1px solid var(--a-border-divider);
}
