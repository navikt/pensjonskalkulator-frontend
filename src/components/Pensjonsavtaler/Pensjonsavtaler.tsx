import { MouseEvent, useContext, useEffect, useMemo, useState } from 'react'
import { FormattedMessage, useIntl } from 'react-intl'
import { useNavigate } from 'react-router'

import { BodyLong, Heading, HeadingProps, Link, VStack } from '@navikt/ds-react'

import { BeregningContext } from '@/pages/Beregning/context'
import { paths } from '@/router/constants'
import {
  useOffentligTpQuery,
  usePensjonsavtalerQuery,
} from '@/state/api/apiSlice'
import {
  generateOffentligTpRequestBody,
  generatePensjonsavtalerRequestBody,
} from '@/state/api/utils'
import { useAppDispatch, useAppSelector } from '@/state/hooks'
import {
  selectAarligInntektFoerUttakBeloep,
  selectAfp,
  selectCurrentSimulation,
  selectEpsHarInntektOver2G,
  selectEpsHarPensjon,
  selectErApoteker,
  selectFoedselsdato,
  selectSamtykke,
  selectSivilstand,
  selectSkalBeregneAfpKap19,
  selectUfoeregrad,
  selectUtenlandsperioder,
} from '@/state/userInput/selectors'
import { userInputActions } from '@/state/userInput/userInputSlice'
import { getFormatMessageValues } from '@/utils/translations'

import ShowMore from '../common/ShowMore/ShowMore'
import { OffentligTjenestepensjon } from './OffentligTjenestePensjon/OffentligTjenestepensjon'
import { PrivatePensjonsavtaler } from './PrivatePensjonsavtaler'

import styles from './Pensjonsavtaler.module.scss'

export const Pensjonsavtaler = ({
  headingLevel,
}: {
  headingLevel: Exclude<HeadingProps['level'], undefined>
}) => {
  const { pensjonsavtalerShowMoreRef } = useContext(BeregningContext)
  const intl = useIntl()
  const navigate = useNavigate()
  const dispatch = useAppDispatch()
  const harSamtykket = useAppSelector(selectSamtykke)
  const sivilstand = useAppSelector(selectSivilstand)
  const aarligInntektFoerUttakBeloep = useAppSelector(
    selectAarligInntektFoerUttakBeloep
  )
  const ufoeregrad = useAppSelector(selectUfoeregrad)
  const afp = useAppSelector(selectAfp)
  const foedselsdato = useAppSelector(selectFoedselsdato)
  const epsHarInntektOver2G = useAppSelector(selectEpsHarInntektOver2G)
  const epsHarPensjon = useAppSelector(selectEpsHarPensjon)
  const erApoteker = useAppSelector(selectErApoteker)
  const utenlandsperioder = useAppSelector(selectUtenlandsperioder)
  const { uttaksalder, aarligInntektVsaHelPensjon, gradertUttaksperiode } =
    useAppSelector(selectCurrentSimulation)
  const skalBeregneAfpKap19 = useAppSelector(selectSkalBeregneAfpKap19)

  const [offentligTpRequestBody, setOffentligTpRequestBody] = useState<
    OffentligTpRequestBody | undefined
  >(undefined)

  const {
    data: offentligTp,
    isLoading: isOffentligTpLoading,
    isError: isOffentligTpError,
  } = useOffentligTpQuery(offentligTpRequestBody as OffentligTpRequestBody, {
    skip: !offentligTpRequestBody || !harSamtykket || !uttaksalder,
  })

  // Hent Offentlig Tjenestepensjon
  useEffect(() => {
    if (harSamtykket && uttaksalder) {
      const requestBody = generateOffentligTpRequestBody({
        afp,
        foedselsdato,
        sivilstand,
        epsHarInntektOver2G,
        epsHarPensjon,
        aarligInntektFoerUttakBeloep: aarligInntektFoerUttakBeloep ?? '0',
        gradertUttak: gradertUttaksperiode ? gradertUttaksperiode : undefined,
        heltUttak: {
          uttaksalder,
          aarligInntektVsaPensjon: aarligInntektVsaHelPensjon,
        },
        utenlandsperioder,
        erApoteker,
      })
      setOffentligTpRequestBody(requestBody)
    }
  }, [harSamtykket, uttaksalder])

  const [pensjonsavtalerRequestBody, setPensjonsavtalerRequestBody] = useState<
    PensjonsavtalerRequestBody | undefined
  >(undefined)

  // Hent Private Pensjonsavtaler
  useEffect(() => {
    if (harSamtykket && uttaksalder) {
      const requestBody = generatePensjonsavtalerRequestBody({
        ufoeregrad,
        afp,
        sivilstand,
        epsHarInntektOver2G,
        epsHarPensjon,
        aarligInntektFoerUttakBeloep: aarligInntektFoerUttakBeloep ?? '0',
        gradertUttak: gradertUttaksperiode ? gradertUttaksperiode : undefined,
        heltUttak: {
          uttaksalder,
          aarligInntektVsaPensjon: aarligInntektVsaHelPensjon,
        },
        skalBeregneAfpKap19,
      })
      setPensjonsavtalerRequestBody(requestBody)
    }
  }, [harSamtykket, uttaksalder])

  const {
    data: pensjonsavtaler,
    isError: isPensjonsavtalerError,
    isSuccess: isPensjonsavtalerSuccess,
  } = usePensjonsavtalerQuery(
    pensjonsavtalerRequestBody as PensjonsavtalerRequestBody,
    {
      skip: !pensjonsavtalerRequestBody || !harSamtykket || !uttaksalder,
    }
  )

  const subHeadingLevel = useMemo(() => {
    return (
      headingLevel ? (parseInt(headingLevel, 10) + 1).toString() : '4'
    ) as Exclude<HeadingProps['level'], undefined>
  }, [headingLevel])

  const onCancel = (e: MouseEvent<HTMLAnchorElement>): void => {
    e.preventDefault()
    dispatch(userInputActions.flush())
    navigate(paths.start)
  }

  const showExplanation =
    (isPensjonsavtalerSuccess &&
      pensjonsavtaler?.avtaler &&
      pensjonsavtaler?.avtaler.length > 0) ||
    (offentligTp?.simuleringsresultatStatus === 'OK' &&
      offentligTp?.simulertTjenestepensjon?.tpNummer !== undefined)

  return (
    <VStack gap="1">
      <Heading id="pensjonsavtaler-heading" level={headingLevel} size="small">
        {intl.formatMessage({ id: 'pensjonsavtaler.title' })}
      </Heading>

      {harSamtykket ? (
        <ShowMore
          ref={pensjonsavtalerShowMoreRef}
          name="pensjonsavtaler"
          aria-labelledby="pensjonsavtaler-heading"
          collapsedHeight={
            (pensjonsavtaler?.avtaler?.length ?? 0) > 1 ? '20rem' : '10rem'
          }
        >
          <>
            <PrivatePensjonsavtaler
              isPartialResponse={!!pensjonsavtaler?.partialResponse}
              isError={isPensjonsavtalerError}
              isSuccess={isPensjonsavtalerSuccess}
              headingLevel={subHeadingLevel}
              privatePensjonsavtaler={pensjonsavtaler?.avtaler}
            />

            <OffentligTjenestepensjon
              isLoading={isOffentligTpLoading}
              isError={isOffentligTpError}
              offentligTp={offentligTp}
              headingLevel={subHeadingLevel}
            />

            {showExplanation && (
              <BodyLong className={styles.footnote}>
                <FormattedMessage id="pensjonsavtaler.fra_og_med_forklaring" />
              </BodyLong>
            )}
          </>
        </ShowMore>
      ) : (
        <BodyLong>
          <FormattedMessage id="pensjonsavtaler.ingress.error.samtykke_ingress" />
          <Link href={paths.start} onClick={onCancel}>
            {intl.formatMessage({
              id: 'pensjonsavtaler.ingress.error.samtykke_link_1',
            })}
          </Link>{' '}
          <FormattedMessage
            id="pensjonsavtaler.ingress.error.samtykke_link_2"
            values={{
              ...getFormatMessageValues(),
            }}
          />
        </BodyLong>
      )}
    </VStack>
  )
}
