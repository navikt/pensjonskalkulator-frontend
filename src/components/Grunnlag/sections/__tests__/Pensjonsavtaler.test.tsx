import { Pensjonsavtaler } from '../Pensjonsavtaler'
import { render, screen } from '@/test-utils'

describe('Pensjonsavtaler', () => {
  it('rendrer ingenting om avtalelisten er tom', () => {
    render(<Pensjonsavtaler pensjonsavtaler={[]} />)

    expect(
      screen.queryByText('Pensjonsavtaler', { exact: false })
    ).not.toBeInTheDocument()
  })

  it('rendrer riktig med avtaler som bare har start dato', () => {
    const { asFragment } = render(
      <Pensjonsavtaler
        pensjonsavtaler={[
          {
            produktbetegnelse: 'DNB',
            kategori: 'privat tjenestepensjon',
            startAlder: 67,
            startMaaned: 1,
            utbetalingsperiode: {
              startAlder: 67,
              startMaaned: 1,
              aarligUtbetaling: 12345,
              grad: 100,
            },
          },
        ]}
      />
    )
    expect(
      screen.queryByText('Pensjonsavtaler', { exact: false })
    ).toBeVisible()
    expect(asFragment()).toMatchSnapshot()
  })

  it('rendrer riktig med avtaler som har bÃ¥de start- og sluttdato', () => {
    const { asFragment } = render(
      <Pensjonsavtaler
        pensjonsavtaler={[
          {
            produktbetegnelse: 'DNB',
            kategori: 'privat tjenestepensjon',
            startAlder: 67,
            startMaaned: 1,
            utbetalingsperiode: {
              startAlder: 67,
              startMaaned: 1,
              sluttAlder: 77,
              sluttMaaned: 8,
              aarligUtbetaling: 12345,
              grad: 100,
            },
          },
        ]}
      />
    )
    expect(
      screen.queryByText('Pensjonsavtaler', { exact: false })
    ).toBeVisible()
    expect(asFragment()).toMatchSnapshot()
  })
})
