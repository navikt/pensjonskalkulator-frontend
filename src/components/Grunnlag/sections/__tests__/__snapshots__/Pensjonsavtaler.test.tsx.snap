// Vitest Snapshot v1

exports[`Pensjonsavtaler > rendrer riktig med avtaler som bare har start dato 1`] = `
<DocumentFragment>
  <div
    class="navds-accordion__item"
    data-testid="pensjonsavtaler"
  >
    <button
      aria-expanded="false"
      class="navds-accordion__header"
      type="button"
    >
      <div
        class="navds-accordion__icon-wrapper"
      >
        <svg
          aria-hidden="true"
          aria-labelledby="title-r0"
          class="navds-accordion__header-chevron"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title-r0"
          >
            Vis mer
          </title>
          <path
            clip-rule="evenodd"
            d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <span
        class="navds-accordion__header-content navds-heading navds-heading--small"
      >
        Pensjonsavtaler: 
        <span
          class="_details_c03039"
        >
          1
        </span>
      </span>
    </button>
    <div
      class="navds-accordion__content navds-accordion__content--closed _content_e53ef6 navds-body-long"
    >
      <table
        class="_tabell_ac6b69"
      >
        <thead>
          <tr>
            <th
              class="_tabellHeader_ac6b69"
            >
              Pensjonsavtale
            </th>
            <th
              class="_tabellHeader_ac6b69 _tabellHeader__Right_ac6b69"
            >
              Årlig utbetaling
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="_tabellCell_ac6b69"
              colspan="2"
            >
              <p
                class="_mellomtittel_ac6b69 navds-body-short"
              >
                Undefined
              </p>
            </td>
          </tr>
          <tr>
            <td
              class="_tabellCell_ac6b69"
            >
              <p
                class="navds-body-short"
              >
                Fra DNB
              </p>
              <p
                class="_utbetaling_ac6b69 navds-body-short"
              >
                Livsvarig utbetaling fra 67 år og 1
              </p>
            </td>
            <td
              class="_tabellCell_ac6b69 _tabellCell__Right_ac6b69"
            >
              <p
                class="navds-body-short"
              >
                1 028,75 kr
              </p>
            </td>
          </tr>
        </tbody>
      </table>
      <p
        class="_paragraph_ac6b69 navds-body-long navds-body-long--small"
      >
        Alle avtaler i privat sektor er hentet fra 
        <a
          class="navds-link"
          href="https://norskpensjon.no/"
        >
          Norsk Pensjon
        </a>
        . Du kan ha andre avtaler enn det som finnes i Norsk Pensjon. Kontakt aktuell pensjonsordning.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Pensjonsavtaler > rendrer riktig med avtaler som har både start- og sluttdato 1`] = `
<DocumentFragment>
  <div
    class="navds-accordion__item"
    data-testid="pensjonsavtaler"
  >
    <button
      aria-expanded="false"
      class="navds-accordion__header"
      type="button"
    >
      <div
        class="navds-accordion__icon-wrapper"
      >
        <svg
          aria-hidden="true"
          aria-labelledby="title-r1"
          class="navds-accordion__header-chevron"
          fill="none"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="title-r1"
          >
            Vis mer
          </title>
          <path
            clip-rule="evenodd"
            d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </div>
      <span
        class="navds-accordion__header-content navds-heading navds-heading--small"
      >
        Pensjonsavtaler: 
        <span
          class="_details_c03039"
        >
          1
        </span>
      </span>
    </button>
    <div
      class="navds-accordion__content navds-accordion__content--closed _content_e53ef6 navds-body-long"
    >
      <table
        class="_tabell_ac6b69"
      >
        <thead>
          <tr>
            <th
              class="_tabellHeader_ac6b69"
            >
              Pensjonsavtale
            </th>
            <th
              class="_tabellHeader_ac6b69 _tabellHeader__Right_ac6b69"
            >
              Årlig utbetaling
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="_tabellCell_ac6b69"
              colspan="2"
            >
              <p
                class="_mellomtittel_ac6b69 navds-body-short"
              >
                Undefined
              </p>
            </td>
          </tr>
          <tr>
            <td
              class="_tabellCell_ac6b69"
            >
              <p
                class="navds-body-short"
              >
                Fra DNB
              </p>
              <p
                class="_utbetaling_ac6b69 navds-body-short"
              >
                Utbetales fra 67 og 1 til 77 år og 8
              </p>
            </td>
            <td
              class="_tabellCell_ac6b69 _tabellCell__Right_ac6b69"
            >
              <p
                class="navds-body-short"
              >
                1 028,75 kr
              </p>
            </td>
          </tr>
        </tbody>
      </table>
      <p
        class="_paragraph_ac6b69 navds-body-long navds-body-long--small"
      >
        Alle avtaler i privat sektor er hentet fra 
        <a
          class="navds-link"
          href="https://norskpensjon.no/"
        >
          Norsk Pensjon
        </a>
        . Du kan ha andre avtaler enn det som finnes i Norsk Pensjon. Kontakt aktuell pensjonsordning.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
